diff --git a/daemon/core/netns/vnode.py b/daemon/core/netns/vnode.py
index 44041cd..4cb5afb 100644
--- a/daemon/core/netns/vnode.py
+++ b/daemon/core/netns/vnode.py
@@ -225,7 +225,11 @@ class SimpleLxcNode(PyCoreNode):
                     "error setting MAC address %s" % str(addr))
     def addaddr(self, ifindex, addr):
         if self.up:
-            self.cmd([IP_BIN, "addr", "add", str(addr),
+            if ":" in str(addr):
+                self.cmd([IP_BIN, "addr", "add", str(addr),
+                  "dev", self.ifname(ifindex)])
+            else:
+                self.cmd([IP_BIN, "addr", "add", str(addr), "broadcast", "+",
                   "dev", self.ifname(ifindex)])
         self._netif[ifindex].addaddr(addr)
 
